feat: Add last snoozed time as a snooze option

vishnu-narayanan opened the pull request:
# Pull Request Template

## Description

Improves UX by allowing users to quickly reuse their last custom snooze time
without having to re-enter it every time.

- Allow last snoozed custom time as an option for conversation snoozing
- Add 'UNTIL_LAST_CUSTOM_TIME' snooze option that shows saved custom time
- Implement smart expiration: times expire when passed OR after 7 days
- Add user-friendly format: 'Sat, 23 Aug, 8.16pm'
- Fix caching issue: actions now update dynamically without page refresh
- Auto-save custom times to localStorage for quick reuse
- Add reactive event system for real-time UI updates
- Include option filtering to show/hide based on saved time availability



<img width="698" height="465" alt="image" src="https://github.com/user-attachments/assets/32edf50a-e717-4bef-8de6-df4d5a80f361" />


## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- Tested locally

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

vishnu-narayanan committed:
feat: add last snoozed time as a snooze option

pranavrajs commented:
Can we do this as well? https://linear.app/chatwoot/issue/CW-4065/feat-ability-to-type-free-form-text-for-custom-snooze

vishnu-narayanan committed:
feat: Add dynamic 'Last snoozed' option with reactive updates

- Add 'UNTIL_LAST_CUSTOM_TIME' snooze option that shows saved custom time
- Implement smart expiration: times expire when passed OR after 7 days
- Add user-friendly format: 'Sat, 23 Aug, 8.16pm'
- Fix caching issue: actions now update dynamically without page refresh
- Auto-save custom times to localStorage for quick reuse
- Add reactive event system for real-time UI updates
- Include option filtering to show/hide based on saved time availability

Improves UX by allowing users to quickly reuse their last custom snooze time
without having to re-enter it every time.

vishnu-narayanan commented:
> Can we do this as well? https://linear.app/chatwoot/issue/CW-4065/feat-ability-to-type-free-form-text-for-custom-snooze

@pranavrajs way outside the scope of this change :P will take up over another PR

vishnu-narayanan committed:
Merge branch 'develop' into feat/add_last_snoozed_option

vishnu-narayanan committed:
Merge branch 'develop' into feat/add_last_snoozed_option

muhsin-k committed:
Merge branch 'develop' into feat/add_last_snoozed_option

muhsin-k committed:
Merge branch 'develop' into feat/add_last_snoozed_option
